### PR TITLE
Build all platforms

### PR DESCRIPTION
**Pull request by gnzlbg**
cc @alexcrichton this needs a more thorough review. It turns out libc failed to build on a lot of older Rust versions for various reasons and platforms, so it took a while to fix the build on all of them.

### CODE REVIEW

**Comment by gnzlbg** [comment]
cc @Susurrus I had to disable some `extra_trait`s in targets in which they were failing to compile (due to too large arrays, `repr(packed)`, etc.). These should be easy to find and fix, i've opened #1243 to track some of these.

File: README.md [hunk 1]
```diff
@@ -8,7 +8,19 @@ Raw FFI bindings to platform libraries like `libc`.
...
 [![Documentation](https://docs.rs/libc/badge.svg)](https://docs.rs/libc)
-![License](https://img.shields.io/crates/l/libc.svg)
+![License](https://img.shields.io/crates/l/libc.svg**
+
+**NOTE:** The minimum supported Rust version is **Rust 1.13.0** . APIs requiring
+newer Rust features are only available on newer Rust versions:
+
+| Feature                 | Version |
+|-------------------------|---------|
+| `union`                |  1.19.0 |
```
Comment (Susurrus) [inline README.md]: I'd recommend manually aligning this table with spaces so it's easier to read un-rendered

Comment (gnzlbg) [reply README.md]: Can do - I hope github uses a monospace font for rendering the diffs.

**Comment by gnzlbg** [comment]
I'm merging this because the `extra_traits` PR broke the library in a couple of platforms and this PR fixes those and ensures that they don't break again. We can iterate on this on master before the next release.

**Pull request merged**